pandas/core/dtypes/dtypes.py

```diff
@@ -1128,6 +1181,15 @@ def __setstate__(self, state):
         # PandasExtensionDtype superclass and uses the public properties to
         # pickle -> need to set the settable private ones here (see GH26067)
         self._subtype = state["subtype"]
+        # backward-compat older pickles won't have "closed" key
+        self._closed = state.pop("closed", None)
+        if self._closed is None:
+            warnings.warn(
+                "Unpickled legacy IntervalDtype does not specify 'closed' "
```

Comment:
test for this?

Reply:
Yes, we have tests for all of the newly-added warning/raising cases in the IntervalDtype constructor.  (just added a missing consistency check https://github.com/pandas-dev/pandas/pull/38394/files#diff-f99ef42afad339d00e36197a60ccc76d74c6a94c30e05aef69d18e0ec4b10d4eR1055)

Reply:
woops, dont have a test for this one.  will update.

Reply:
huh, this is definitely reached, but tm.assert_produces_warning isnt seeing anything